vertical-pod-autoscaler/pkg/recommender-external/input/recommendations_feeder.go

```diff
@@ -69,36 +71,92 @@ func NewExternalRecommendationsStateFeeder(config *rest.Config, clusterState *up
 }
 
 func (e externalRecommendationsStateFeeder) LoadVPAs() {
+	// Add or update existing VPAs in the model.
+	vpaKeys := make(map[upstream_model.VpaID]bool)
```

Comment:
since you are using the map as a set to check existence: `make(map[upstream_model.VpaID]struct{})` and later assign an empty struct: `vpaKeys[vpa.ID] = struct{}{}`
This saves memory when sets containers a lot of elements, and also we understand that you use a set and nobody will ask what is the meaning of the boolean value.

Reply:
done